File: GXpert/GXpert.Web/Modules/Content/ContentTopic/ContentTopicColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Content.Columns;

[ColumnsScript("Content.ContentTopic")]
[BasedOnRow(typeof(ContentTopicRow), CheckNames = true)]
public class ContentTopicColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [QuickFilter]
    public string ContentTitle { get; set; }
    [QuickFilter]
    public string ClassTitle { get; set; }
    [QuickFilter]
    public string SubjectTitle { get; set; }
    [QuickFilter]
    public string TopicTitle { get; set; }
    [QuickFilter]
    public string MediumTitle { get; set; }
    public short SortOrder { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentTopicColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContentTopicRow } from "./ContentTopicRow";

export interface ContentTopicColumns {
    Id: Column<ContentTopicRow>;
    ContentTitle: Column<ContentTopicRow>;
    ClassTitle: Column<ContentTopicRow>;
    SubjectTitle: Column<ContentTopicRow>;
    TopicTitle: Column<ContentTopicRow>;
    MediumTitle: Column<ContentTopicRow>;
    SortOrder: Column<ContentTopicRow>;
}

export class ContentTopicColumns extends ColumnsBase<ContentTopicRow> {
    static readonly columnsKey = 'Content.ContentTopic';
    static readonly Fields = fieldsProxy<ContentTopicColumns>();
}